🐛 PagerDuty Slack on-call aborted when it couldn't map a user

## Expected Behavior

When a PagerDuty user can't be mapped to a Slack user, it should reference them by their PagerDuty name (https://github.com/ministryofjustice/data-platform/blob/main/scripts/pagerduty/slack-on-call/main.py#L48-L60)

## Actual Behavior

An error from Slack API causes the script to exit

```python
Traceback (most recent call last):
  File "/home/runner/work/data-platform/data-platform/scripts/pagerduty/slack-on-call/main.py", line 70, in <module>
    main()
  File "/home/runner/work/data-platform/data-platform/scripts/pagerduty/slack-on-call/main.py", line 59, in main
    if get_slack_user_id() is None:
  File "/home/runner/work/data-platform/data-platform/scripts/pagerduty/slack-on-call/main.py", line 49, in get_slack_user_id
    response = slack_client.users_lookupByEmail(email=get_on_call_user()[1])
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/slack_sdk/web/client.py", line 45[22](https://github.com/ministryofjustice/data-platform/actions/runs/5022173064/jobs/9005331962#step:8:23), in users_lookupByEmail
    return self.api_call("users.lookupByEmail", http_verb="GET", params=kwargs)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/slack_sdk/web/base_client.py", line 156, in api_call
    return self._sync_send(api_url=api_url, req_args=req_args)
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/slack_sdk/web/base_client.py", line 187, in _sync_send
    return self._urllib_api_call(
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/slack_sdk/web/base_client.py", line [30](https://github.com/ministryofjustice/data-platform/actions/runs/5022173064/jobs/9005331962#step:8:31)9, in _urllib_api_call
    return SlackResponse(
  File "/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/slack_sdk/web/slack_response.py", line 199, in validate
    raise e.SlackApiError(message=msg, response=self)
slack_sdk.errors.SlackApiError: The request to the Slack API failed. (url: https://www.slack.com/api/users.lookupByEmail)
The server responded with: {'ok': False, 'error': 'users_not_found'}
Error: Process completed with exit code 1.
```

## Steps to Reproduce the Problem

  1. Run [`.github/workflows/platform-pagerduty-slack-on-call.yml`](https://github.com/ministryofjustice/data-platform/blob/main/.github/workflows/platform-pagerduty-slack-on-call.yml) with a user the script can't map

## Other information

The engineer on-call was @tom-webber, Tom's email in PagerDuty is his MoJ Digital, but Slack is DOM1, therefore the script can't map, but should use the plaintext version

--- a/scripts/pagerduty/slack-on-call/README.md
+++ b/scripts/pagerduty/slack-on-call/README.md
@@ -27,7 +27,7 @@ cd scripts/pagerduty/slack-on-call
 
 # Run Python container
 docker run -it --rm \
-  --entrypoint /bin/sh \
+  --entrypoint /bin/bash \
   --volume $( pwd ):/app \
   --workdir /app \
   --env PAGERDUTY_SCHEDULE_ID="REPLACE_ME" \

--- a/scripts/pagerduty/slack-on-call/main.py
+++ b/scripts/pagerduty/slack-on-call/main.py
@@ -46,11 +46,13 @@ def get_on_call_user():
 
 
 def get_slack_user_id():
-    response = slack_client.users_lookupByEmail(email=get_on_call_user()[1])
     user_id = None
 
-    if response["ok"]:
+    try:
+        response = slack_client.users_lookupByEmail(email=get_on_call_user()[1])
         user_id = response["user"]["id"]
+    except Exception:
+        pass
 
     return user_id
 

--- a/scripts/pagerduty/slack-on-call/requirements.txt
+++ b/scripts/pagerduty/slack-on-call/requirements.txt
@@ -1,2 +1,2 @@
-pdpyras==4.5.2
+pdpyras==5.0.3
 slack-sdk==3.21.3

